diff --git a/src/libFLAC/metadata_iterators.c b/src/libFLAC/metadata_iterators.c
index 35137496..8b2f582a 100644
--- a/src/libFLAC/metadata_iterators.c
+++ b/src/libFLAC/metadata_iterators.c
@@ -3602,7 +3602,15 @@ FLAC__bool get_file_stats_(const char *filename, struct flac_stat_s *stats)
        FLAC__ASSERT(0 != stats);
        return (0 == flac_stat(filename, stats));
 }
+#ifdef __PSP__

+void set_file_stats_(const char *filename, struct flac_stat_s *stats)
+{
+    (void)filename;
+    (void)stats;
+}
+
+#else
 void set_file_stats_(const char *filename, struct flac_stat_s *stats)
 {
 #if defined(_POSIX_C_SOURCE) && (_POSIX_C_SOURCE >= 200809L) && !defined(_WIN32)
@@ -3629,6 +3637,7 @@ void set_file_stats_(const char *filename, struct flac_stat_s *stats)
        FLAC_CHECK_RETURN(chown(filename, -1, stats->st_gid));
 #endif
 }
+#endif

 int fseek_wrapper_(FLAC__IOHandle handle, FLAC__int64 offset, int whence)
 {
